File: generator/src/test/resources/kotlin/expected/sunday/BaseUriTest/test-baseurl-generation-in-api.output.kt
package io.test.service

import io.outfoxx.sunday.MediaType
import io.outfoxx.sunday.Operation
import io.outfoxx.sunday.OperationSpec
import io.outfoxx.sunday.Transport
import io.outfoxx.sunday.URITemplate
import io.outfoxx.sunday.http.Method
import io.outfoxx.sunday.http.Request
import io.outfoxx.sunday.operation
import io.test.Environment
import kotlin.String
import kotlin.Unit
import kotlin.collections.List

public class API<Req : Request>(
  public val transport: Transport<Req>,
  public val defaultContentTypes: List<MediaType> = listOf(),
  public val defaultAcceptTypes: List<MediaType> = listOf(MediaType.JSON),
) {
  public fun fetchTest(): Operation<Unit, String, Req> =
      this.transport.operation<Unit, String, Req>(
    OperationSpec(
      method = Method.Get,
      pathTemplate = "/tests",
      acceptTypes = this.defaultAcceptTypes
    )
  )

  public companion object {
    public fun baseURL(
      server: String = "master",
      environment: Environment = Environment.Sbx,
      version: String = "1",
    ): URITemplate = URITemplate(
      "http://{server}.{environment}.example.com/api/{version}",
      mapOf("server" to server, "environment" to environment, "version" to version)
    )
  }
}

File: generator/src/test/resources/swift/expected/BaseUriTest/test-baseurl-generation-in-api.output.swift
import Sunday

public final class API<TransportType : Transport> : Sendable {

  public static var problemTypes: [ProblemRegistration] {
    return []
  }
  public let transport: TransportType
  public let defaultContentTypes: [MediaType]
  public let defaultAcceptTypes: [MediaType]

  public init(
    transport: TransportType,
    defaultContentTypes: [MediaType] = [],
    defaultAcceptTypes: [MediaType] = [.json],
    problemTypes: [ProblemRegistration] = API.problemTypes
  ) {
    self.transport = transport
    self.defaultContentTypes = defaultContentTypes
    self.defaultAcceptTypes = defaultAcceptTypes
    problemTypes.forEach { $0.register(on: transport) }
  }

  public static func baseURL(
    server: String = "master",
    environment: Environment = Environment.sbx,
    version: String = "1"
  ) -> URI.Template {
    return URI.Template(
      format: "http://{server}.{environment}.example.com/api/{version}",
      parameters: [
        "server": server,
        "environment": environment,
        "version": version
      ]
    )
  }

  public func fetchTest() throws -> Sunday.Operation<Empty, String, TransportType> {
    return Sunday.Operation(
      transport: self.transport,
      spec: Sunday.OperationSpec(
        method: .get,
        pathTemplate: "/tests",
        pathParameters: nil,
        queryParameters: nil,
        body: Empty.none,
        contentTypes: nil,
        acceptTypes: self.defaultAcceptTypes,
        headers: nil
      )
    )
  }

}
